index.d.ts

```diff
@@ -17,6 +17,14 @@ export type PromiseFactory<ValueType> = () => PromiseLike<ValueType>;
 
 export type ProgressNotifier = (progress: number) => void;
 
+export type SettledPromise<ResolveValueType = unknown> = {
+	status: 'fulfilled';
+	value: ResolveValueType;
+} | {
+	status: 'rejected';
+	reason: Error;
+};
```

Comment:
Could we borrow the type from `Promise.allSettled()`? If not, at least ensure the type is identical to what's used there.

Reply:
The TypeScript typings for `Promise.allSettled()` don't actually export this object type so we are forced to re-define this object.

https://github.com/microsoft/TypeScript/blob/582e404a1041ce95d22939b73f0b4d95be77c6ec/lib/lib.es2020.promise.d.ts#L21-L50